Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5316,6 +5324,8 @@ void IFileContentManagerHost.ReportContent(FileArtifact artifact, in FileMateria
                     existence = PathExistence.ExistsAsFile;
                 }
 
+                MakeSharedOpaqueOutputIfNeeded(artifact.Path);

```

Comment:
Here, every time content is reported, if it corresponds to an output file and the file exists we apply the timestamps if that path is under an output directory

Reply:
Just double checking. Are you sure ReportContent is always called? At least for the shared opaque case, there were cases where pips were failing in an early stage and stuff was not being reported yet, but files were already produced. There is a comment under FlagSharedOpaque... about using the raw data that comes from detours that contains dynamic writes, that was because of this.

---
In reply to: [272298907](https://github.com/Microsoft/BuildXL/pull/83#discussion_r272298907) [](ancestors = 272298907)

Reply:
That's why there is another place where this is called. That other place is `FlagSharedOpaqueOutputs`, which is called from `PostProcess`